### PR TITLE
Moving all asset register/enqueue to Assets.php via Service_Provider

### PR DESCRIPTION
**Pull request by Camwyn**
This utilizes the service provider in common, and moves all asset handling to `src/Tribe/Assets.php`

Ticket: https://central.tri.be/issues/83302

### CODE REVIEW

File: src/Tribe/Assets.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+	/**
+	 * Enqueue scripts for admin views
+	 *
+	 * @since TBD
+	 */
+	public function admin_enqueue_scripts() {
+		global $post;
+
+		$modules = apply_filters( 'tribe_events_tickets_modules', null );
```
Comment (bordoni) [inline src/Tribe/Assets.php]: Add Docblock.

File: src/Tribe/Assets.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	 * @since TBD
+	 */
+	public function admin_enqueue_scripts() {
+		global $post;
+
+		/**
+		 * Filter the array of module names.
+		 *
+		 * @since TBD
+		 *
```
Comment (bordoni) [inline src/Tribe/Assets.php]: Add `@param`.

File: src/Tribe/Assets.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+							'data' => array(
+								'decimal' => $decimal,
+								'decimal_error' => __( 'Please enter in without thousand separators and currency symbols.', 'event-tickets' ),
+							),
+						),
+					),
+				)
+			);
+
+			wp_enqueue_script( 'tribe-bumpdown' );
```
Comment (bordoni) [inline src/Tribe/Assets.php]: Why not add this as a dependency of something?

File: src/Tribe/Assets.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		global $post;
+
+		/**
+		 * Filter the array of module names.
+		 *
+		 * @since TBD
+		 *
+		 * @param array the array of modules
+		 *
+		 * See event-tickets/src/Tribe/Tickets.php->modules()
```
Comment (barryhughes) [inline src/Tribe/Assets.php]: Optional - but we could make this into a `@see` tag

File: src/Tribe/Assets.php [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+		 */
+		$modules = apply_filters( 'tribe_events_tickets_modules', null );
+
+		// For the metabox
+		if ( ! empty( $post ) &&
```
Comment (barryhughes) [inline src/Tribe/Assets.php]: Let's bump the first part of the conditional to the next line so we're following [this standard](http://moderntribe.github.io/products-engineering/php/#php-control-multi) - we should also move hte `&&` operator to the start of the line that follows - ie:

```php
if (
	$first
	&& $second
	&& $third
) {
	// Body
}
```

File: src/Tribe/Assets.php [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+			 ! empty( $modules ) &&
```
Comment (barryhughes) [inline src/Tribe/Assets.php]: Ditto, move `&&` to the start of the next line

File: src/Tribe/Assets.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+			$locale  = localeconv();
+			$decimal = isset( $locale['decimal_point'] ) ? $locale['decimal_point'] : '.';
+
+			/**
+			 * Filter the decimal point character used in the price
+			 * @param string $decimal the decimal character to filter
+			 *
+			 * @since TBD
+			 *
+			 * @param array the array of modules
```
Comment (barryhughes) [inline src/Tribe/Assets.php]: C+P error? Looks like this should probably be `@param string $decimal`

Comment (Camwyn) [inline src/Tribe/Assets.php]: Yes, y'know I can't find anywhere we _add_ this filter, so 🤞 it's a string 😉

**Pull request merged**